tests/testthat/test-fb_count_species_by_traits.R

```diff
@@ -0,0 +1,57 @@
+# Fake dataset ----
+
+species_traits <- data.frame(
+  species = paste0("sp", 1:4),
+  t1      = c(NA, 2.5, 100, 400),
+  t2      = c(2.2, 5.0, 200, 200)
+)
+
+
+test_that("fb_count_species_by_traits() errors with wrong inputs", {
+  
+  # Wrong inputs ----
+  
+  expect_error(
+    fb_count_species_by_traits(),
+    "Argument 'species_traits' (species x traits data frame) is required",
+    fixed = TRUE
+  )
+  
+  expect_error(
+    {
+      st2 <- species_traits
+      colnames(st2) <- NULL
+      fb_count_species_by_traits(st2)
+    },
+    "The species x traits object must have column names (trait names)",
+    fixed = TRUE
+  )
+  
+  expect_error(
+    fb_count_species_by_traits(species_traits[ , -1, drop = FALSE]),
+    "The species x traits object must contain the 'species' column",
+    fixed = TRUE
+  )
+  
+  
+  # Success ----
```

Comment:
This part should be moved in another `test_that({})` because this mentions at L10 that it's to test that the function errors with wrong inputs.